flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -469,18 +475,57 @@ public void sqlUpdate(String stmt) {
 				createTableOperation.getCatalogTable(),
 				createTableOperation.getTableIdentifier(),
 				createTableOperation.isIgnoreIfExists());
+		} else if (operation instanceof CreateDatabaseOperation) {
+			CreateDatabaseOperation createDatabaseOperation = (CreateDatabaseOperation) operation;
+			catalogManager.createDatabase(createDatabaseOperation.getCatalogName(),
+										createDatabaseOperation.getDatabaseName(),
+										createDatabaseOperation.getCatalogDatabase(),
+										createDatabaseOperation.isIgnoreIfExists(),
+										false);
 		} else if (operation instanceof DropTableOperation) {
 			DropTableOperation dropTableOperation = (DropTableOperation) operation;
 			catalogManager.dropTable(
 				dropTableOperation.getTableIdentifier(),
 				dropTableOperation.isIfExists());
-		} else if (operation instanceof UseCatalogOperation) {
-			UseCatalogOperation useCatalogOperation = (UseCatalogOperation) operation;
-			catalogManager.setCurrentCatalog(useCatalogOperation.getCatalogName());
+		} else if (operation instanceof DropDatabaseOperation) {
+			DropDatabaseOperation dropDatabaseOperation = (DropDatabaseOperation) operation;
+			catalogManager.dropDatabase(
+					dropDatabaseOperation.getCatalogName(),
+					dropDatabaseOperation.getDatabaseName(),
+					dropDatabaseOperation.isIfExists(),
+					dropDatabaseOperation.isRestrict(),
+					false);
+		} else if (operation instanceof AlterDatabaseOperation) {
+			AlterDatabaseOperation alterDatabaseOperation = (AlterDatabaseOperation) operation;
+			catalogManager.alterDatabase(
+					alterDatabaseOperation.getCatalogName(),
+					alterDatabaseOperation.getDatabaseName(),
+					alterDatabaseOperation.getCatalogDatabase(),
+					false);
+		} else if (operation instanceof UseOperation) {
+			applyUseOperation((UseOperation) operation);
 		} else {
 			throw new TableException(
 				"Unsupported SQL query! sqlUpdate() only accepts a single SQL statements of " +
-					"type INSERT, CREATE TABLE, DROP TABLE, USE CATALOG");
+					"type INSERT, CREATE TABLE, DROP TABLE, USE CATALOG, USE [catalog.]database, " +
+					"CREATE DATABASE, DROP DATABASE, ALTER DATABASE");
```

Comment:
With the increasing length of the supported operations, maybe we should switch to saying something like "xxx is not supported".

Reply:
Yes, but now we can't infer the type of unknown. We can do a refactor later.